Project architecture/MVVM.md

```diff
@@ -110,28 +152,68 @@ open class BaseViewModel<State : Any, Event : Any> : ViewModel() {
+    // Update the UI state
+}
 
-loginViewModel.viewEventData().observe(this, { event ->
-	// Handle event
-})
+loginViewModel.viewEventData().observe(this) { event ->
+    // Handle event
+}
 
 ```
 
 This is now enough to establish a connection between the View and ViewModel.
 
 So far, we've only concentrated on the View and ViewModel, but what about the Model in MVVM? The Model is responsible for managing the data received from a specific data source (Database, Network, etc.), completely UI independent. This means that the Model should expose its data only to the ViewModel. The ViewModel can also request some data from the Model, so it is a two-way communication.
 
+### Implementation using StateFlow and SharedFlow
+
+If we want to use StateFlow and SharedFlow as a substitution to the LiveData objects mentioned above, our BaseViewModel will look something like this:
+
+```kotlin
+abstract class BaseViewModel<State : Any, Event : Any>(private val initialState: State) : ViewModel() {
+
+    val state: StateFlow<State> get() = stateFlow
+    val commonState: StateFlow<CommonState> get() = commonStateFlow
+    val event: SharedFlow<Event> get() = eventFlow
+
+    private val stateFlow: MutableStateFlow<State> by lazy { MutableStateFlow(initialState) }
+    private val commonStateFlow = MutableStateFlow(CommonState())
+    private val eventFlow: MutableSharedFlow<Event> = MutableSharedFlow()
+}
+```
+
+The private properties `stateFlow`, `commonStateFLow`, `eventFlow` are mutable only within the BaseViewModel and we expose them through the properties `state`, `commonState`, `event` which are immutable.
+
+A backing field and function would be really helpful for using these objects inside our ViewModels which inherit from the BaseViewModel, therefore, in our BaseViewModel we will add the following code:
+
+```kotlin
+abstract class BaseViewModel<State : Any, Event : Any>(private val initialState: State) : ViewModel() {
+
+    //...
+
+    var viewState: State
+        get() = state.value
+        set(state) {
+            stateFlow.value = state
+        }
+
+    protected suspend fun emitEvent(event: Event) {
+        eventFlow.emit(event)
+    }
+}
+```
+
+In the Activity or Fragment, we will handle the states and events and adjust the UI accordingly.
+
 At this point, the concept of MVVM and how to implement it should be much clearer and easier to understand. Take into account that the code you saw was very simplified, just to show the conceptual idea of MVVM in Android. This is only one way to implement it, there are many more different kinds of implementation that may include [DataBinding](https://developer.android.com/topic/libraries/data-binding/) or are very [Rx](http://reactivex.io/) heavy.
 
 A textbook example of our proposal of MVVM architecture can be found on this [GitHub repo](https://github.com/infinum/Android-MvvM-Example). In the README, you will find all the details for a base setup and using Android architecture components.
```

Comment:
@ffloreani Kristina mentioned this in the PR description. Here is a link to the private Infinum repo in the public Handbook. Do we keep it?

Reply:
Sorry for taking so long to answer 😅 yeah, we should not have any links from the public to the private handbook. It breaks our SEO and page rank on Google. We can just remove the link and we should be good 😄

Reply:
I removed it. 👍 I also updated some links in the MVP chapter which were linking existing chapters, but I guess some restructuring of the handbook was done in the meantime, so the links were giving me “This chapter does not exist.” error even when I was logged in with Infinum ID. One more thing 😅 , in the Mobile app architecture chapter, in the Documentation section, there are some links to keynote presentations on Infinum’s Google drive which are not available to the public (only the link to the PDF template is accessible), so I guess parts of that section should be removed as well. We could consider relocating this section somewhere in the private repo chapters because it seems useful.

Reply:
That sounds like a good idea to me

Reply:
I'd really love to keep the _Mobile app architecture_ page public, it's good content that can attract other devs to Infinum 👿 

A fun fact is that links that are hidden behind Google's own login are not counted as broken links 😄 so no worries about things stored on our Google Drive

Reply:
I meant moving parts of the Documentation section to the private repo (the steps with instructions “How to” looked very Infinum specific to me), definitely didn’t think about moving the whole chapter. 😅 However, yeah, since they don’t count as broken links, no biggie, they might as well stay. 😄 Make the devs come to Infinum if they want to see where the links go? 😉